crates/sui-indexer-alt-framework/src/mocks/store.rs

```diff
@@ -287,35 +282,88 @@ impl MockStore {
             self.commit_failures.failures - prev
         );
 
-        // Store the data
-        let mut total_count = 0;
-        {
-            let mut data = self.data.lock().unwrap();
-            for (checkpoint, checkpoint_values) in values {
-                total_count += checkpoint_values.len();
-                data.entry(checkpoint)
-                    .or_default()
-                    .extend(checkpoint_values);
+        let key = pipeline.to_string();
+        let mut total = 0;
+        if let Some(inner) = self.data.get(&key) {
+            for (cp, v) in values {
+                total += v.len();
+                inner.entry(cp).or_insert_with(Vec::new).extend(v);
             }
+            return Ok(total);
+        }
+
+        // create once, then insert
+        let inner = DashMap::new();
+        for (cp, v) in values {
+            total += v.len();
+            inner.entry(cp).or_insert_with(Vec::new).extend(v);
+        }
+        self.data.insert(key, inner);
```

Comment:
The logic seems duplicative here. Can we first insert into `data` if an entry doesn't exist for a key, and then just use the same logic? Something like
```rust
let inner = self.data.entry(key).or_insert_with(DashMap::new);
for (cp, v) in values {
    total += v.len();
    inner.entry(cp).or_insert_with(Vec::new).extend(v);
}

```
Same for `commit_data`.